Assets/HoloToolkit/UX/Scripts/Tooltip/ToolTipConnector.cs

```diff
@@ -0,0 +1,350 @@
+                    toolTip.transform.eulerAngles = eulerAngles;
+                    switch (PivotingMode)
+                    {
+                        case ConnnectorPivotMode.Automatic:
+                            Transform relativeTo = null;
+                            switch (PivotDirectionOrient)
+                            {
+                                case ConnectorOrientType.OrientToCamera:
+                                    relativeTo = Camera.main.transform;//Veil.Instance.HeadTransform;
+                                    break;
+
+                                case ConnectorOrientType.OrientToObject:
+                                    relativeTo = Target.transform;
+                                    break;
+                            }
+                            Vector3 localPosition = GetDirectionFromPivotDirection(PivotDirection, ManualPivotDirection, relativeTo) * pivotDistance;
+                            toolTip.PivotPosition = Target.transform.position + localPosition;
+                            break;
+
+                        case ConnnectorPivotMode.Manual:
+                            // Do nothing
+                            break;
+                    }
+                    break;
+
+                case ConnectorFollowType.XRotation:
+                    // Set the transform of the entire tool tip
+                    // Set the pivot relative to target/camera
+                    toolTip.transform.position = Target.transform.position;
+                    toolTip.transform.rotation = Target.transform.rotation;
+                    switch (PivotingMode)
+                    {
+                        case ConnnectorPivotMode.Automatic:
+                            Transform relativeTo = null;
+                            switch (PivotDirectionOrient)
+                            {
+                                case ConnectorOrientType.OrientToCamera:
+                                    relativeTo = Camera.main.transform;//Veil.Instance.HeadTransform;
+                                    break;
+
+                                case ConnectorOrientType.OrientToObject:
+                                    relativeTo = Target.transform;
+                                    break;
+                            }
+                            toolTip.PivotPosition = Target.transform.position + GetDirectionFromPivotDirection(
+                                PivotDirection,
+                                ManualPivotDirection,
+                                relativeTo) * pivotDistance;
+                            break;
+
+                        case ConnnectorPivotMode.Manual:
+                            // Do nothing
+                            break;
+                    }
+                    break;
+            }
+        }
+
+        private void Update()
+        {
+            UpdatePosition();
+        }
+
+        private void OnDrawGizmos ()
```

Comment:
No need for space before ()

Are gizmos for more than just editor mode?

Reply:
No, they're not, but this Unity API isn't called unless the application is running in the editor.